pattern/constraint/thing/RelationConstraint.java

```diff
@@ -94,20 +94,14 @@ public void addTo(GraphTraversal.Thing traversal) {
         for (RolePlayer rolePlayer : rolePlayers) {
             assert !rolePlayer.inferredRoleTypes.isEmpty();
             ThingVariable player = rolePlayer.player();
-            int rep = rolePlayer.repetition();
             if (rolePlayer.roleType().isPresent() && rolePlayer.roleType().get().id().isName()) {
-                Identifier.Scoped role = Identifier.Scoped.of(owner.id(), rolePlayer.roleType().get().id(), player.id(), rep);
+                Identifier.Scoped role = Identifier.Scoped.of(owner.id(), rolePlayer.roleType().get().id(), player.id(), rolePlayer.repetition());
                 traversal.relating(owner.id(), role);
                 traversal.playing(player.id(), role);
                 traversal.isa(role, rolePlayer.roleType().get().id());
                 traversal.types(role, rolePlayer.inferredRoleTypes());
-            } else if (rep > 1) {
-                Identifier.Scoped role = Identifier.Scoped.of(owner.id(), null, player.id(), rep);
-                traversal.relating(owner.id(), role);
-                traversal.playing(player.id(), role);
-                traversal.types(role, rolePlayer.inferredRoleTypes());
```

Comment:
we no longer need to de-optimise parallel edges since the traversal engine handles the more general case:

ie. `(friend: $x, friend: $x)` can be treated as two optimised traversal edges, because the traversal has to handle the case where `(friend: $x, friend: $y)` concidentally ends up with `$x` and `$y` being the same concept.